### PR TITLE
Bugfix FXIOS-10777 pull refresh easter egg doesn't cancel when refresh is cancelled

### PR DESCRIPTION
**Pull request by FilippoZazzeroni**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-10777)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/23508)

## :bulb: Description
Bug fix Pull refresh easter egg getting showed when pull refresh is cancelled.
Bug fix leak in pull refresh caused by `listenForThemChanges`

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [x] If needed, I updated documentation / comments for complex code and public methods
- [x] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Comment by FilippoZazzeroni** [comment]
@mergify backport release/v134

File: firefox-ios/Client/Frontend/Browser/TabScrollController.swift [hunk 1]
```diff
@@ -174,20 +178,19 @@ class TabScrollingController: NSObject,
...
-                guard let self else { return }
-                // If we are in the home page delete pull to refresh so it doesn't show on the background
-                if tabIsLoading() || (tab?.isFxHomeTab ?? false) {
-                    pullToRefreshView?.stopObservingContentScroll()
-                    pullToRefreshView?.removeFromSuperview()
-                } else {
-                    configureRefreshControl(isEnabled: true)
-                }
-            }
+    private func onTabLoading() {
```
Comment (Cramsden) [inline firefox-ios/Client/Frontend/Browser/TabScrollController.swift:188]: I think this function name doesn't really tell me what this code is doing. Is there a method name that would potentially be more descriptive of what is happening here?

File: firefox-ios/Client/Frontend/Browser/WebView/PullRefreshView.swift [hunk 2]
```diff
@@ -224,6 +233,7 @@ class PullRefreshView: UIView,
     }
 
     deinit {
+        easterEggTimer?.cancel()
```
Comment (Cramsden) [inline firefox-ios/Client/Frontend/Browser/WebView/PullRefreshView.swift:236]: do we also want to nil out the timer here?

File: firefox-ios/Client/Frontend/Browser/TabScrollController.swift (hunk 1 above)
Comment (FilippoZazzeroni) [reply firefox-ios/Client/Frontend/Browser/TabScrollController.swift:188]: Yes i agree i'm modifying it

**Review by PARAIPAN9 — APPROVED** [review]
Tested, LGTM just the thing that @Cramsden mentioned I can say I agree with it. It is a good practice to `nil` the `DispatchSourceTimer` in de `deinit`.

**Comment by FilippoZazzeroni** [comment]
Thanks so much @PARAIPAN9. I'll nilled out same for the scroll observer

**Pull request merged**